jupyterhub_files/user_data_worker.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/bash -ex
+exec 1> >(logger -s -t $(basename $0)) 2>&1 # Redirect stdout/stderr to the syslog
+
+# Disable SSH service so the spawner cannot connect during this process.
+# If an error occurs or the script exits successfully, SSH should be restarted automatically.
+trap "service ssh restart" EXIT
+service ssh stop
+
+# Setup mount point
+mkdir -p /jupyteruser
+
+# Mount EBS home volume if a device is specified
+if [ -n "{device}" ]; then
```

Comment:
I notice that this if construct does not have an else branch to specify what should happen if the condition is false, i.e., if a device is not specified. Naturally, if a device is not specified, the statements in the if construct will be skipped and the statements after the construct will be executed. I presume that that is the intended behavior, namely, if a device is not specified, the user will be set up on the root volume. If that is the case, then I wonder whether it might be better to include an else branch on the if construct with a [:](https://stackoverflow.com/questions/17583578/what-command-means-do-nothing-in-a-conditional-in-bash) statement - just to make it explicit to a reader of the code that if a device is not mounted, we want the shell to simply move on to execute the statements after the if construct. In other words, without an else branch, a reader might wonder whether, say, an exception should be raised if a device is not specified, and whether the statements following the if construct should or should not be executed. Using an else "do nothing" clears up the confusion, I think. Thoughts?

Reply:
@joshuagetega Thanks for the feedback. I think it's a good idea to make it clear to the reader what should be done if no device is given for the user home directory. In that case, we want to make it clear that the root device should be used for the home directory and user account setup. I've updated the `else` branch of that conditional with a no-op and a comment to that effect. Take a look and let me know if it looks OK.